src/server/editorServices.ts

```diff
@@ -16,6 +16,11 @@ namespace ts.server {
     }
 
     const lineCollectionCapacity = 4;
+    const getCanonicalFileName = createGetCanonicalFileName(sys.useCaseSensitiveFileNames);
+    const globalCachePath = toPath(
+        ".typingCache",
+        process.env[(process.platform === "win32") ? "USERPROFILE" : "HOME"],

```

Comment:
This is environment specific, as it depends on the global `process` being available, which is a Node.js thing. Please abstract this (probably via `ServerHost` which uses `System` from `sys.ts`).